src/miner.cpp

```diff
@@ -176,7 +176,8 @@ CBlockTemplate* CreateNewBlock(const CScript& scriptPubKeyIn)
                     if (!mempool.mapTx.count(txin.prevout.hash))
                     {
                         LogPrintf("ERROR: mempool transaction missing input\n");
-                        if (fDebug) assert("mempool transaction missing input" == 0);
+			// the following assert is dead code (comparing a string literal to 0)
+                        // if (fDebug) assert("mempool transaction missing input" == 0);
```

Comment:
I think the intent was, if `fDebug` is true, then fail with the message "mempool transaction missing input". That will happen because `assert()` prints out the condition that failed, and `"mempool transaction missing input" == 0` will always fail.